lib/StaticUniformGrid.js

```diff
@@ -0,0 +1,477 @@
+        var items = cellData.items;
+        if (itemCount > 0) {
+            cellIndices.push(firstFreeIndex);
+        } else {
+            cellIndices.push(-1);
+        }
+        cellCounts.push(itemCount);
+        firstFreeIndex += itemCount;
+        for (var j = 0; j < itemCount; ++j) {
+            cellItems.push(clone(items[j]));
+        }
+    }
+}
+
+defineProperties(StaticUniformGrid.prototype, {
+    /**
+     * Gets the width of the uniform grid's cells.
+     * @memberof StaticUniformGrid.prototype
+     * @type {Number}
+     */
+    cellWidth : {
+        get : function() {
+            return this._cellWidth;
+        }
+    },
+    /**
+     * Gets the axis aligned bounding box around the grid.
+     * @memberof StaticUniformGrid.prototype
+     * @type {AxisAlignedBoundingBox}
+     */
+    axisAlignedBoundingBox : {
+        get : function() {
+            return this._axisAlignedBoundingBox;
+        }
+    },
+    /**
+     * Gets cell resolution in x, y, and z.
+     * @memberof StaticUniformGrid.prototype
+     * @type {Cartesian3}
+     */
+    resolution : {
+        get : function() {
+            return this._resolution;
+        }
+    },
+    /**
+     * Gets the list of indices at which each cell's data starts in the data array.
+     * @memberof StaticUniformGrid.prototype
+     * @type {Object[]}
+     */
+    cellIndices : {
+        get : function() {
+            return this._cellIndices;
+        }
+    },
+    /**
+     * Gets the list of lengths in the data array for each cell's data.
+     * @memberof StaticUniformGrid.prototype
+     * @type {Object[]}
+     */
+    cellCounts : {
+        get : function() {
+            return this._cellCounts;
+        }
+    },
+    /**
+     * Gets the data array that all the geometry/spatial data is stored in.
+     * @memberof StaticUniformGrid.prototype
+     * @type {Object[]}
+     */
+    data : {
+        get : function() {
+            return this._data;
+        }
+    }
+});
+
+function CellBucket() {
+    this.itemCount = 0;
+    this.items = [];
+}
+
+function BinningParameters(allCellData, cellCheckFunction, grid) {
+    this.allCellData = allCellData;
+    this.cellCheckFunction = cellCheckFunction;
+    this.grid = grid;
+    this.item = {};
+}
+
+var cellMarchScratch = new Cartesian3();
+/**
+ * Function for executing a function on each cell touched by an axis aligned bounding box within the uniform grid.
+ *
+ * @param {StaticUniformGrid} [grid] A StaticUniformGrid to be marched over
+ * @param {Cartesian3} [min] The minimum coordinate of the bounding box to be "marched" over.
+ * @param {Cartesian3} [max] The maximum coordinate of the bounding box to be "marched" over.
+ * @param {Function} [marchFunction] A function to apply at each cell. The function should expect as inputs its parameters, the cell's minimum coordinate, and the cell width.
+ * @param {Object} [parameters] Parameters for the marchFunction
```

Comment:
All these options are required, so remove the square brackets. (Also in other areas in this file)